### PR TITLE
water usage model refactor

### PR DESCRIPTION
**Pull request by wasswarichard**
Fixes migrations issues and refactors water usage calculator component

### CODE REVIEW

File: packages/api/db/migration/20221118081755_alter_irrigation_task_table.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ export const down = async function (knex) {
 
   for (const row of task_rows) {
     await knex.raw(
-      `UPDATE 'irrigation_task' task SET hours = ${
+      `UPDATE "task" t SET duration = ${
         row.duration / 60
```
Comment (Duncan-Brain) [inline packages/api/db/migration/20221118081755_alter_irrigation_task_table.js:49]: @wasswarichard 

Is the previous state of task.duration = irrigation_task.duration/60 or is the previous state NULL? 
image

**Comment by Duncan-Brain** [comment]
@wasswarichard 

I think it mostly looks ok to me. I added one inline comment on the alter table.

Just one question -- since this migration has already been merged. And is probably running on beta. Is there a process to rollback the previous db update and re-run the migration?

**Comment by wasswarichard** [comment]
@Duncan-Brain the previous state is not deployed to beta, I rolled back.

**Review by Duncan-Brain — APPROVED** [review]
I think that's good.

**Comment by anant00729** [comment]
The only change I can see is you are setting the duration = null in the update query and there is a modal name change in the files.
This looks good to me.

**Review by anant00729 — APPROVED** [review]

**Pull request merged**